### DISCUSSION TITLE
Getting PRs for Private/Org repos to list?

### DISCUSSION THREAD

**Discussion by tstackhouse** [lowlighter/metrics#232] [Q&A] [answered]
Been trying to get my metrics to show the PR's I've created/reviewed in my private org, and I cannot for the life of me figure out what I'm doing wrong:

https://github.com/tstackhouse/tstackhouse/blob/main/.github/workflows/metrics.yml

My token has `repo` ans `read:org` permissions, and it's got SSO auth. It seems like it should be simple enough, but I feel like I'm just missing something.

Update: Did some more looking into this, and poking at the GraphQL API, it looks like in order to get org repos that you have access to requires both `affiliations` and `ownerAffiliations` properties in the query, see [this thread](https://github.community/t/graphql-api-how-to-fetch-all-repositories-that-the-current-user-has-access-to/13792)

**Comment by lowlighter** [comment, marked as answer]
Sorry to hear that you had trouble with setup 😕 
I don't use GitHub organizations that much so I may be missing something too.

The PR reviews/created stats are obtained from `contributionsCollection` from the following query:
https://github.com/lowlighter/metrics/blob/d9c943f724f66c262f0104dc440590c85a237b61/source/plugins/base/queries/user.graphql#L34-L41

It isn't affected by the `affiliations` constraint, so maybe it is possible that this endpoint actually doesn't consider organizations whatever scopes are granted? GraphQL explorer indicates that there's an option `organizationID` but not sure what's it's doing.
If that's the case maybe we'll need to switch away and find another solution to get these 😕 

I didn't know the `affiliations` trick, that's a really nice find 😮 

It may actually fix other metrics in this case since as you may have noticed some stats are computed by iterating on repositories (like stargazers, forks, watchers and languages). 

https://github.com/lowlighter/metrics/blob/d9c943f724f66c262f0104dc440590c85a237b61/source/plugins/base/index.mjs#L26

By adding `, affiliations: ${affiliations.toLocaleUpperCase()}` next to `, ownerAffiliations: ${affiliations.toLocaleUpperCase()}` in the line above, it may result in more accurate stats for these.

___

If you want you can fork this repo, make changes on it and then switch used action from `uses: lowlighter/metrics@latest` to `uses: tstackhouse/metrics@master` so you can troubleshoot more easily.

Only drawback is that the docker image will be rebuilt each time on forks, so workflows will be around ~5min which may be a bit tedious for quick tests.

I'll try to troubleshoot from my side too whenever I have some free time, hope it'll help 🙂

**Reply by tstackhouse** [reply]
I'll give it a try when I get a chance! I came across this while I was looking at someone's profile today and decided to give it a whirl :)

**Reply by lowlighter** [reply]
I may have some news on this 🙂 

Searching around, it seems that `contributionsCollection` depends on `read:user` scope instead of `repo` scope like other private repositories metrics.

```js
> node graphql.mjs
{
  totalRepositoriesWithContributedCommits: 31,
  totalCommitContributions: 1655,
  restrictedContributionsCount: 0,
  totalIssueContributions: 41,
  totalPullRequestContributions: 126,
  totalPullRequestReviewContributions: 23
}
> node graphql.mjs
{
  totalRepositoriesWithContributedCommits: 18,
  totalCommitContributions: 1410,
  restrictedContributionsCount: 0,
  totalIssueContributions: 39,
  totalPullRequestContributions: 126,
  totalPullRequestReviewContributions: 20
}
```

**Reply by tstackhouse** [reply]
I went to go try out this change in a fork and you beat me to it! :D

Giving my fork a whirl all the same, since I already switched over to it in my readme repo

**Reply by tstackhouse** [reply]
It's definitely something with the scope of the token, I ran that same query in the api explorer and got this:

```
{
  "data": {
    "viewer": {
      "contributionsCollection": {
        "totalRepositoriesWithContributedCommits": 7,
        "totalCommitContributions": 1947,
        "restrictedContributionsCount": 2,
        "totalIssueContributions": 10,
        "totalPullRequestContributions": 195,
        "totalPullRequestReviewContributions": 438
      }
    }
  }
}
```
